File: back_end/MindEarth/MindEarth.Web/Features/Roles/DTO/DTO_Role.cs
namespace MindEarth.Web.Features.Roles.DTO
{
    public class DTO_Role
    {
        public string? RowId { get; set; } = null!;

        public string Text { get; set; } = null!;


        public bool IsActive { get; set; }

        public List<DTO_RoleDetail> Roles  { get; set; } = new List<DTO_RoleDetail>();
    }

    public class DTO_RoleDetail
    {
        public string? RowId { get; set; } = null!;
        public string MenuDetailId { get; set; }

        public bool AddRight { get; set; }
        public bool ViewRight { get; set; }
        public bool DeleteRight { get; set; }
        public bool ModifyRight { get; set; }

    }
}

File: admin_ui/report_admin_ui/src/app/features/auth/user-controller/models/dto-role.ts
export class DtoRole {
    rowId?: string | null;
    text?: string;
    isActive?: boolean;
    roles?: DTO_RoleDetail[];
}

export class DTO_RoleDetail
{
    rowId?: string | null;
    menuDetailId?: string;
    addRight?: boolean;
    viewRight?: boolean;
    deleteRight?: boolean;
    modifyRight?: boolean;
}